### PR TITLE
Prevent initialize from being called multiple times

### LINKED ISSUE #267: Handle initialize() being called multiple times with same trackingId
Prevent `ga('create', X)` being called multiple times for same trackingId and warn user when it happens.

### LINKED ISSUE #399: Script tag gets inserted twice to DOM on ReactGA.initialize()
After initializing the script with `ReactGA.initialize("UA-XXXXXXXXX-X");`, the following script tag can be found in the DOM twice: 

```html
<script async="" src="https://www.google-analytics.com/analytics.js"></script>
```

Screenshot 2019-11-21 at 16 13 20

### PR DESCRIPTION
**Pull request by copperwall**
This prevents the google analytics tag from being loaded more than once when `ReactGA.initialize` is called.

Now if client code calls `ReactGA.initialize('UA-fasdf')` multiple times, the GA tag will only be added once, and a `create` impression will only be sent once.

Closes #399 
Closes #267

### CODE REVIEW

**Comment by antondomratchev** [comment]
+1. I was about to open an issue with the problem that this PR solves. Would love to see this merged in for the next release.

**Comment by JClackett** [comment]
Please merge this! :)

**Comment by copperwall** [comment]
Oops, I didn't realize this broke all of the tests 😬 I'll see what I can do about those

**Comment by antondomratchev** [comment]
@copperwall I believe most of the breakages are because each test is not cleaning up. I was thinking of adding an `afterEach` that sets `window.ga = null` which might only work in cases where init was called. Most likely this will require reworking some tests to rely on a `beforeEach` setup call that would init the tracker.
just my 2c

**Comment by bigblobby** [comment]
Any progress on this?

**Comment by copperwall** [comment]
sorry this kind of fell off my plate, I can take a stab at the tests again soon

File: package.json [hunk 1]
```diff
@@ -89,8 +89,8 @@
     "react": "^15.6.2",
     "react-addons-test-utils": "^15.0.1",
     "react-dom": "^15.0.1",
-    "react-router": "4.4.0",
-    "react-router-dom": "4.4.0",
```
Comment (SimeonC) [inline package.json:93]: Not sure why this is on a beta version over the non rc version?

Comment (SimeonC) [reply package.json:93]: Eh, since when did they un-publish versions 😨

File: src/utils/loadGA.js [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
   } else if (options && options.debug) {
     gaAddress = 'https://www.google-analytics.com/analytics_debug.js';
   }
 
+  return gaAddress;
+}
+
+function loadGA(options) {
+  const gaAddress = optionToGaAddress(options);
+
```
Comment (SimeonC) [inline src/utils/loadGA.js:16]: I don't think we need a memoize function, we only support one `ga` being loaded so just doing a boolean here would be fine.
```
let isLoaded = false;
function loadGA(options) {
  if (isLoaded) return;
  ...
```

**Comment by copperwall** [comment]
@SimeonC Thank you for the review! I can make some changes this weekend

**Comment by SimeonC** [comment]
@copperwall sorry this has a bunch of conflicts now 😅 

Good news is that whenever you can get this updated it'll auto-release without me procrastinating for weeks 😆 

Hopefully, this should also encourage me to actually review things faster as I can do it all on github now 🚀

File: src/utils/loadGA.js (hunk 2 above)
Comment (copperwall) [reply src/utils/loadGA.js]: Ah thanks, that makes this PR a lot less complicated. It seems to play a lot more nicely with the test suite as well.

**Comment by copperwall** [comment]
@SimeonC sorry it took so long to get these changes in. I fixed the merge conflicts and dialed back the memoization change and added the boolean change like you suggested.

If there's anything else this needs I'm happy to help!

**Comment by SimeonC** [comment]
Looks good to me, thanks!

**Comment by copperwall** [comment]
Thanks for the quick response!

**Pull request merged**